packages/frontend-shared/src/gql-components/HeaderBarContent.vue

```diff
@@ -245,7 +246,7 @@ const userData = computed(() => {
 })
 
 const savedState = computed(() => {
-  return props.gql?.currentProject?.savedState
+  return props.gql?.currentProject?.savedState as AllowedState
```

Comment:
Curious why we cannot get the types via GraphQL (didn't look too closely)

Reply:
The field is currently defined this way, so it appears to be deliberate

```
t.json('savedState', {
      description: 'Project saved state',
      resolve: (source, args, ctx) => {
        return ctx.project.getCurrentProjectSavedState()
      },
    })
```

Perhaps this is a safety measure since theoretically the content on disk can be manually mutated in ways that cause a schema mismatch?

Reply:
Not sure, maybe you cannot type `t.json` (limitation).

We can find out more at some point, had a quick look but seems like it's not a simple "just add the types" change and this PR is getting fairly beefy.